src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/HiviUap.java

```diff
@@ -0,0 +1,52 @@
+package com.wikia.webdriver.pageobjectsfactory.componentobject.ad;
+
+import com.wikia.webdriver.common.core.WikiaWebDriver;
+import com.wikia.webdriver.common.core.elemnt.Wait;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+public class HiviUap {
+
+  private final Wait wait;
+  private final String slot;
+
+  private static final String SLOT_SELECTOR = "#%s ";
+  private static final String REPLAY_SELECTOR = ".replay-overlay";
+  private static final String VIDEO_CLICK_AREA_SELECTOR = ".toggle-ui-overlay";
+  private static final String VIDEO_THUMBNAIL_SELECTOR = ".videoThumbnail";
+
+  private final WikiaWebDriver driver;
+
+  public HiviUap(WikiaWebDriver driver, String slot) {
+    this.wait = new Wait(driver);
+    this.driver = driver;
+    this.slot = slot;
+  }
+
+  public void waitForVideoStart() {
+    usingAdFrame(() -> {
+      wait.forElementNotVisible(By.cssSelector(VIDEO_THUMBNAIL_SELECTOR));
+    });
+  }
+
+  public void waitForVideoEnd() {
+    usingAdFrame(() -> {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove useless curly braces around statement (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1602)